### ISSUE TITLE
Erroneous metadata

### ISSUE THREAD

**Issue by mduering** [tomahawk-player/tomahawk#378]
Hello,

I understand the unique concept behind Tomahawk and enjoy exploring its functionalities. But I wonder how Tomahawk retrieves data/metadata for albums: On a few occasions I noticed that song titles were completely wrong and seemingly taken randomly from other albums but not from my collection. (The albums appear correctly in other players.) My collection is mostly based on self-digitized CDs stored as .flac files on a Synology NAS. It seems that T prefers to look up links for these tracks elsewhere - if this is correct, can I change it to play the files on my NAS? Can I ensure it prioritizes the metadata I have already?

In a nutshell, I want T to play my collection and suggest other interesting titles to me + additional info on the artists.

And a small bug: On OSX, the artist bio page does not scroll.

Many thanks,
Marten

**Comment by jherskowitz** [comment]
We currently do track listing lookups using a number of online sources (e.g. Musicbrainz/Last.fm).  We plan on using local data if available in a future release.

The bio is also a known issue at the moment (all of older tickets are on a JIRA instance we are working on recovering).  Thanks for the report.

**Comment by theli-ua** [comment]
@mduering could you provide example of an album where tomahawk presents you with erroneous metadata?

**Comment by theli-ua** [comment]
Sometimes tomahawk will show me a random tracklist for particular release(s) , though restart fixes it for me :-/  
Not sure how would I go about reproducing this issue for fixing though

**Comment by albertocalmeida** [comment]
I usually get the same bug but only regarding the YEAR of an album. I tag my mp3s with Musicbrainz Picard which uses similar online sources, but Tomahawk gets that property wrong quite often so far as to changing it from track to track within one album (not compilation). 
I can easily provide a list:
Nevermore - Dead Heart In A Dead World
Nevermore - Enemies Of Reality
Metallica - Ride The Lightning
Metallica - Master Of Puppets 
Dream Theater - Images And Words
Dream Theater - Falling To Infinity
Opeth - Blackwater Park
Opeth - Deliverance

This could be a very long list... Hope it helps!

**Comment by mduering** [comment]
Hi all,

first of all my apologies for taking so long to respond and many thanks for getting back to me so quickly! 
Anton, you asked for examples of erroneous metadata: I first noticed this with Blues Brothers albums. Live in Montreux, the Soundtrack and a Best of album all appeared scrambled. On other albums, individual songs appear as greyed out. I assume that the online database have a different track listing than the album I have. If the title does not match it appears as grey and the local file is not displayed.

Hope this is helpful,
Marten

**Comment by theli-ua** [comment]
Once again got wrong tracklist for an album.
Was opening Phantoms album by Freezepop, but got tracklist from this: https://en.wikipedia.org/wiki/Phantom_Planet_%28album%29

**Comment by theli-ua** [comment]
Fri Jan 15 2016 - 14:44:05 [1]: View page shown: "Phantoms" 
Fri Jan 15 2016 - 14:44:05 [1]: virtual void Tomahawk::InfoSystem::MusicBrainzPlugin::notInCacheSlot(Tomahawk::InfoSystem::InfoStringHash, Tomahawk::InfoSystem::InfoRequestData) "http://musicbrainz.org/ws/2/release?query=release:"Phantoms" AND artist:"Freezepop"&limit=100" 
Fri Jan 15 2016 - 14:44:05 [9]: Rovi request url: "http://api.rovicorp.com/search/v2/music/search?query=Freezepop Phantoms&entitytype=album&include=album:tracks&apikey=7jxr9zggt45h6rg2n4ss3mrj&sig=67b29fedf2da2ac7b37bc30222a3c236" 
Fri Jan 15 2016 - 14:44:05 [8]: bool ViewHeader::checkState() Giving columns of view "" initial weighting: (1) for 1 columns 
Fri Jan 15 2016 - 14:44:05 [1]: void Tomahawk::InfoSystem::MusicBrainzPlugin::gotReleasesSlot() "<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
<metadata xmlns="http://musicbrainz.org/ns/mmd-2.0#" xmlns:ext="http://musicbrainz.org/ns/ext#-2.0" created="2016-01-15T11:37:18.450Z">
 <release-list offset="0" count="0"/>
</metadata>
" 
Fri Jan 15 2016 - 14:44:05 [8]: Doing spotify album lookup via webservice with ID: "spotify:album:38rv3SC9DX09kHCXaW3Ijc" 
Fri Jan 15 2016 - 14:44:06 [1]: void Tomahawk::Pipeline::checkQIDState(const query_ptr&)   "a5829273-bdfd-4766-a229-cd93fa894005"   1 
Fri Jan 15 2016 - 14:44:06 [1]: void PlayableModel::finishLoading() 
Fri Jan 15 2016 - 14:44:06 [1]: virtual bool PlayableModel::removeRows(int, int, const QModelIndex&) 0 0 QModelIndex(-1,-1,0x0,QObject(0x0) ) 
Fri Jan 15 2016 - 14:44:06 [8]: Enqueueing command to thread: Tomahawk::DatabaseWorker(0x7fc1a00013c0) 0 "loadsocialactions" 
Fri Jan 15 2016 - 14:44:06 [9]: virtual void Tomahawk::DatabaseCommand_LoadSocialActions::exec(Tomahawk::DatabaseImpl*) 
Fri Jan 15 2016 - 14:44:06 [8]: Dispatching to resolver "DatabaseResolver" 0 "Query(a5829273-bdfd-4766-a229-cd93fa894005, Freezepop - The Happy Ending on Phantoms)" false "a5829273-bdfd-4766-a229-cd93fa894005"

**Comment by theli-ua** [comment]
Weird thing is that it got the right album on spotify https://play.spotify.com/album/38rv3SC9DX09kHCXaW3Ijc   , not sure where it got the tracklist from

**Comment by goncalor** [comment]
I always edit the metadata for my collections. In Tomahawk about 97% of my collection shows an empty Year column though.

**Comment by LoveTatFitties** [comment]
I really love Tomahawk, but this issue is driving me insane!! [This guy is staring at me](http://www.last.fm/music/Seacat/+images/3a1526ab4f4d4cccb0e7658583e66175) and he is not a member of the band nor is the album cover. I have the correct album cover on every mp3 I've paid for but for some reason, Tomahawk refuses to show them and instead it shows a picture that some random Russian guy uploaded a year before the band was created.

I am begging you on my knees, please fix this

**Comment by muesli** [comment]
Dupe of #443.
